code/datums/abilities/flock/flockmind.dm

```diff
@@ -353,21 +351,30 @@
 	if(locate(/obj/flock_structure) in T)
 		boutput(holder.owner, "<span class='alert'>There is already a flock structure on this flocktile!</span>")
 		return 1
+
 	for (var/atom/O in T.contents)
 		if (O.density && !isflock(O))
 			boutput(holder.owner, "<span class='alert'>That tile has something that blocks tealprint creation!</span>")
 			return 1
+
+	var/list/friendlyNames = list()
+	var/mob/living/intangible/flock/flockmind/F = holder.owner
+	for(var/obj/flock_structure/S as anything in F.flock.unlockedStructures)
+		friendlyNames += initial(S.flock_id) //flock_id is the friendly name for the structure
+
+
 	//todo: replace with FANCY tgui/chui window with WHEELS and ICONS and stuff!
-	var/structurewanted = tgui_input_list(holder.owner, "Select which structure you would like to create", "Tealprint selection", list("Collector", "Sentinel"))
+	var/structurewanted = tgui_input_list(holder.owner, "Select which structure you would like to create", "Tealprint selection", friendlyNames)
+
 	if (!structurewanted)
 		return TRUE
-	switch(structurewanted)
-		if("Collector")
-			structurewantedtype = /obj/flock_structure/collector
-			resourcecost = 200
-		if("Sentinel")
-			structurewantedtype = /obj/flock_structure/sentinel
-			resourcecost = 300
+	var/obj/flock_structure/structurewantedtype = null
+	for(var/obj/flock_structure/S as anything in F.flock.unlockedStructures)
```

Comment:
```suggestion
	for(var/S in F.flock.unlockedStructures)
```
`S` is also a weird name, maybe change to `structure_type` since literal type paths are very unclear in DM.